### DISCUSSION TITLE
KMS Wallet Client

### DISCUSSION THREAD

**Discussion by tuler** [wevm/viem#668] [Idea / Feature Request]
Are there any plans, or is there any third-party support for a KMS based WalletClient?
Does it make sense?

Similar to [eth-signer-kms](https://github.com/rumblefishdev/eth-signer-kms) for ethers.js or [ethers-signers](https://docs.rs/ethers-signers/latest/ethers_signers/) for rust.

**Comment by jxom** [comment]
Not right now. Wouldn't be hard to make a [Custom Account](https://viem.sh/docs/accounts/custom.html) for it though.

**Reply by Andriy-Kulak** [reply]
if there are any examples of how to do this properly, feel free to post! would love to use viem for KMS if possible
